irods/test/collection_test.py

```diff
@@ -380,6 +398,77 @@ def test_object_paths_with_dot_and_dotdot__323(self):
         home2 = normalize('/zone','holmes','..','home','public','..','user')
         self.assertEqual(home2, '/zone/home/user')
 
+    def test_update_mtime_of_collection_using_touch_operation_as_non_admin__525(self):
+        user_session = self.logins.session_for_user(RODSUSER)
+
+        # Capture mtime of the home collection.
+        home_collection_path = helpers.home_collection(user_session)
+        collection = user_session.collections.get(home_collection_path)
+        old_mtime = collection.modify_time
+
+        # Set the mtime to an earlier time.
+        user_session.collections.touch(home_collection_path, seconds_since_epoch=1400000000)
+
+        # Compare mtimes for correctness.
+        collection = user_session.collections.get(home_collection_path)
+        self.assertGreater(old_mtime, collection.modify_time)
+
+    def test_touch_operation_does_not_create_new_collections__525(self):
+        user_session = self.logins.session_for_user(RODSUSER)
+
+        # The collection should not exist.
+        collection_path = f'{helpers.home_collection(user_session)}/test_touch_operation_does_not_create_new_collections__525'
+        with self.assertRaises(CollectionDoesNotExist):
+            user_session.collections.get(collection_path)
+
+        # Show the touch operation throws an exception if the target collection
+        # does not exist.
+        with self.assertRaises(CollectionDoesNotExist):
+            user_session.collections.touch(collection_path)
+
+        # Show the touch operation did not create a new collection.
+        with self.assertRaises(CollectionDoesNotExist):
+            user_session.collections.get(collection_path)
+
+    def test_touch_operation_only_works_for_collections_as_non_admin__525(self):
```

Comment:
I think both are correct, but what you've provided is clearer. I don't think `collection_manager` needs to be included as that's kind of an implementation detail. I think it's safe to assume the test is about the collection-based implementation of touch since we're in the collections test script.

So, will change to `test_touch_does_not_work_when_given_a_data_object__525`.

Reply:
Oh... I got the impression from the name that an admin *could* provide a data object to the `collection_manager.touch` operation and it would work, so that's why I suggested the alternative name. Is that the case?

Reply:
> Is that the case?

No. The touch op for the collection mgr and data object mgr aren't interchangeable. I disallowed that due to the PRC's design.

Reply:
Done.